core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/V4Properties.java

```diff
@@ -66,4 +65,65 @@ public boolean shouldDoubleUrlEncode() {
     public boolean shouldNormalizePath() {
         return normalizePath;
     }
+
+    public static Builder builder() {
+        return new BuilderImpl();
+    }
+
+    public interface Builder {
+        Builder credentials(AwsCredentialsIdentity credentials);
+
+        Builder credentialScope(CredentialScope credentialScope);
+
+        Builder signingClock(Clock signingClock);
+
+        Builder doubleUrlEncode(boolean doubleUrlEncode);
+
+        Builder normalizePath(boolean normalizePath);
+
+        V4Properties build();
+    }
+
+    private static class BuilderImpl implements Builder {
+        private AwsCredentialsIdentity credentials;
+        private CredentialScope credentialScope;
+        private Clock signingClock;
+        private boolean doubleUrlEncode;
+        private boolean normalizePath;
+
+        @Override
+        public Builder credentials(AwsCredentialsIdentity credentials) {
+            this.credentials = Validate.paramNotNull(credentials, "Credentials");
+            return this;
+        }
+
+        @Override
+        public Builder credentialScope(CredentialScope credentialScope) {
+            this.credentialScope = Validate.paramNotNull(credentialScope, "CredentialScope");
+            return this;
+        }
+
+        @Override
+        public Builder signingClock(Clock signingClock) {
+            this.signingClock = signingClock;
+            return this;
+        }
+
+        @Override
+        public Builder doubleUrlEncode(boolean doubleUrlEncode) {
+            this.doubleUrlEncode = doubleUrlEncode;
+            return this;
+        }
+
+        @Override
+        public Builder normalizePath(boolean normalizePath) {
```

Comment:
Should we use Boolean to differentiate between false and unset? Same for doubleUrlEncode

Reply:
It always need to be set. Guess I forgot to add the not-null validation, done.

Reply:
It'll never be null though if we use boolean instead of Boolean